### PR TITLE
Bootstrap improvements

### PR DESCRIPTION
**Pull request by inverse**
## Description

- Leverage `.python-version` file to manage default Python version - tools such as mise/pyenv cal also leverage this
  - Leverage it in CI
- Migrate bootstrap logic to script for easier reuse
- Add `.envrc`  to automate venv loading 

---

## Related Issues

- Related to #136

---

## Type of Changes
Mark the type of changes included in this pull request:

- [ ] Bugfix
- [ ] New Feature
- [ ] Documentation Update
- [ ] Refactor
- [x] Other (please specify):

---

## Checklist
- [ ] I have tested the changes locally and they work as expected.
- [ ] I have added/updated necessary documentation (if applicable).
- [ ] I have followed the code style and guidelines of the project.
- [ ] I have searched for and linked any related issues.

---

## Additional Notes
Add any additional comments, screenshots, or context for the reviewer(s).

---

Thank you for your contribution to PyTado! 🎉

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -59,11 +59,8 @@
     },
     "image": "mcr.microsoft.com/vscode/devcontainers/python:3.12",
     "name": "PyTado",
-    "postCreateCommand": "python3 -m venv venv && . venv/bin/activate && pip install --upgrade pip && pip install -e '.[all]' && pre-commit install",
-    "updateContentCommand": "python3 -m venv venv && . venv/bin/activate && pip install --upgrade pip && pip install -e '.[all]' && pre-commit install",
-    "runArgs": [
```
Comment (inverse) [inline .devcontainer/devcontainer.json:64]: Should be reverted and fixed with #136

File: .envrc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export VIRTUAL_ENV=."venv"
+layout python
+
+[[ -f .envrc.private ]] && source_env .envrc.private
```
Comment (inverse) [inline .envrc:4]: Allows custom private environment variables to be defined such as TADO username/pass for easier testing.

File: .envrc [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export VIRTUAL_ENV=."venv"
```
Comment (inverse) [inline .envrc:1]: I saw dev containers script used `.venv` name so made it consistent - happy to change it the other way to `venv` if you'd rather.

Comment (wmalgadey) [reply .envrc:1]: I do most of my dev work out of vscode and vscode creates the .venv dir

File: .envrc (hunk 2 above)
Comment (wmalgadey) [reply .envrc:4]: we should add this to the examples and readme

File: .envrc (hunk 3 above)
Comment (inverse) [reply .envrc:1]: AHH sorry this one is wrong, yeah feels like with the leading . Is the normal one. Will update this and remove the other change when I get home

**Pull request merged**